accepted/0000-caasp-integration.md

```diff
@@ -0,0 +1,102 @@
+- Feature Name: CaaS Platform support
+- Start Date: 2019-09-24
+- RFC PR: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Initial integration for managing a CaaS Platform cluster from Uyuni/SUSE Manager.
+
+# Motivation
+[motivation]: #motivation
+
+Uyuni/SUSE Manager is the systems management tool and must offer to the end-user the possibility to manage other SUSE products like CaaS Platform.
+Uyuni/SUSE Manager already supports Kubernetes cluster integration. It works with CaaS Platform version 3 (container engine: Docker).
+
+In this regard, this RFC will cover the following aspects:
+
+- Node status reporting
+- Update the current Kubernetes integration of Uyuni/SUSE Manager to work with CaaS Platform version 4 while still supporting CaaS Platform version 3
+- Document how to deploy CaaS Platform using AutoYaST
+
+# Detailed design
+[design]: #detailed-design
+
+## Node status reporting
+
+All the nodes in the cluster can be registered to Uyuni/SUSE Manager using [bootstrapping methods](https://opensource.suse.com/doc-susemanager/suse-manager/client-configuration/registration-overview.html).
+This brings two major benefits:
+
+- The user can see from Uyuni/SUSE Manager if a node has packages updates and patch available, along with all the information that a Salt minion is presenting when registered to Uyuni/SUSE Manager (e.g. CVE auditing)
+- Another benefit is that the user can associate staged channels to the nodes registered (along with an activation key) to associate channels to the nodes of the cluster.
+
+The agent `skuba-update`, running on each node of the cluster, automatically picks up the packages from the associated channels and proceed to automatically update the nodes.
+NOTE: `skuba-update` is **NOT** invoked by Uyuni/SUSE Manager and it is running autonomously on the nodes of the CaaS Platform cluster.
+
+For example, an organization could have a "staging" cluster with a set of staging channels. The operator would cherry-pick Kubernetes updates from the SCC channels into the staging ones and then wait for skuba-update to roll them out into the staging cluster. At a later time, once testing is green inside of the staging cluster, the packages can be promoted to the "production channels" and be rolled out to the production cluster.
+
+It is paramount to state in the user interface and the documentation that the information presented for the minion is in read-only mode.
+If a user reboots the node, installs a patch or installs a package related to Kubernetes (`kubernetes-kubeadm kubernetes-kubelet kubernetes-client cri-o cni-plugins`) the Kubernetes may break.
```

Comment:
How does Uyuni 'prevent stupid' here ?

Reply:
For the initial iteration, we thought of a UI message and documentation. In the next iterations, we might want to consider an approach like minion blackout.

Reply:
:+1: for ui message and documentation. I'm not convinced that minion blackout is the right thing.